docs/source/onedal/algorithms/ensembles/decision-forest.rst

```diff
@@ -0,0 +1,218 @@
+(:math:`\mathrm{observationsPerTreeFraction}`), and the number of features to try as a possible split per
+node (:math:`\mathrm{featuresPerNode}`), the algorithm does the following:
+
+#. For each tree (:math:`1, \ldots, \mathrm{M}`):
+#. Generate a bootstrapped set of observations with :math:`\mathrm{observationsPerTreeFraction} * |S|`
+   elements in it.
+#. Start with the tree whose depth is equal to :math:`0`.
+#. For each leaf node in the tree:
+    - Randomly choose :math:`\mathrm{featuresPerNode}` features to be tried as possible splits.
+    - Find the best split that maximizes :math:`{I}_{\mathrm{reduction}}`.
+    - Split current node into two based the best split.
+#. Stop when a termination criterion is met.
+#. Update OOB error metric (optional).
+#. Update feature importance metric (optional).
+
+The library supports the following termination criteria to stop growing the tree:
+
+- Minimal number of observations in a leaf node. Node :math:`t` is not processed if the subset of
+  observations is smaller than the predefined value. Splits that produce nodes with the number of
+  observations smaller than that value are not allowed.
+- Maximal tree depth. Node :math:`t` is not processed if its depth in the tree reaches the predefined
+  maximal value.
+- Impurity threshold. Node :math:`t` is not processed if its :math:`I` value is smaller than the predefined
+  minimal value defined with parameter :math:`\mathrm{impurityThreshold}`.
+
+.. _df_t_math_dense:
+
+Training method: *Dense*
+~~~~~~~~~~~~~~~~~~~~~~~~
+In "Dense" training method all possible split variants for each feature (from selected features' subset for current node) are evaluated 
+for best split computation.
+
+.. _df_t_math_hist:
+
+Training method: *Hist*
+~~~~~~~~~~~~~~~~~~~~~~~~
+"inexact" (also called "histogram") training method. In this method we consider only some selected subset of splits for best split computation. 
+This subset of splits is computed for each feature on initialization stage of the algorithm. After computing subset of splits, we substitute 
+each value from initially provided data with the value of the corresponding bin. Bins are continuous intervals between selected splits.
+
+.. _df_i_math:
+.. _df_i_math_dense:
+.. _df_i_math_hist:
+
+Inference methods: *Dense* and *Hist*
+-------------------------------------
+"Dense" and "Hist" inference methods performs prediction by the same way:
+
+#. For classification, :math:`y_i \in \{0, \ldots, \mathrm{C-1}\}`, where :math:`C` is the number of classes,
+   the tree ensemble model predicts the output by selecting the response :math:`y`,
+   which is voted for by the majority of the trees in the forest.
+
+#. For regression, the tree ensemble model uses the mean of :math:`M` functions' results to predict the
+   output, i.e. :math:`\hat{y}=\frac{1}{M} \sum_{k=1}^M{f_k(x_i)}, \; f_k \in F,` where :math:`F=\{f : \mathbb{R}^p \rightarrow W, \; W \subset \mathbb{R}, \; |W|=T\}` is a set of 
```

Comment:
I'd suggest to remove `F=\{f : \mathbb{R}^p \rightarrow W, \; W \subset \mathbb{R}, \; |W|=T\}`. It adds unnecessary complexity. 

`where f_k are regression tree ...` is enough.

Reply:
fixed